RecoLocalCalo/HcalRecAlgos/src/MahiFit.cc

```diff
@@ -292,7 +292,6 @@ void MahiFit::updatePulseShape(double itQ, FullSampleVector &pulseShape, FullSam
       double tmp = 0.5*( nnlsWork_.pulseP[iTS+delta]*nnlsWork_.pulseP[jTS+delta] +
 			 nnlsWork_.pulseM[iTS+delta]*nnlsWork_.pulseM[jTS+delta] );
 
-      pulseCov(iTS+nnlsWork_.maxoffset,jTS+nnlsWork_.maxoffset) += tmp;
```

Comment:
i think that this should be skipped/removed only for  ```iTS==jTS``` the off-diagonal parts still need to be filled

Reply:
Thanks for pointing this out. You are correct and I just committed a fix.